R/iSEE-main.R

```diff
@@ -311,9 +318,13 @@ iSEE <- function(
     }
 
     observeEvent(input$tour_firststeps, {
-        intro_firststeps <- read.delim(system.file("extdata", "intro_firststeps.txt",package = "iSEE"),
-                                       sep=";", stringsAsFactors = FALSE,row.names = NULL)
-        introjs(session, options = list(steps= intro_firststeps))
+        if(is.null(tour)) {
+          intro_firststeps <- read.delim(system.file("extdata", "intro_firststeps.txt",package = "iSEE"),
+                                         sep=";", stringsAsFactors = FALSE,row.names = NULL)
+          introjs(session, options = list(steps = intro_firststeps))
+        } else {
+          introjs(session, options = list(steps = tour))
+        }
```

Comment:
Or you could just assign `tour <- intro_firststeps` and avoid duplicating the `introjs` call.

Reply:
or even better, to not even use the `intro_firststeps` variable and setting `tour` to `read.delim(system.file("extdata", "intro_firststeps.txt",package = "iSEE"), stringsAsFactors = FALSE,row.names = NULL)`

Reply:
yes.